libs/sdk-ui-kit/src/Dialog/HubspotConversionTouchPointDialogBase.tsx

```diff
@@ -0,0 +1,190 @@
+) => {
+    const {
+        targetId,
+        hubspotPortalId,
+        hubspotFormId,
+        selectedValue,
+        cancelButtonText,
+        dialogTitle,
+        showCancelButton,
+        values = {},
+        onClose,
+        onFormSubmitted,
+    } = props;
+
+    const intl = useIntl();
+    const [isFormReady, setIsFormReady] = useState(false);
+    const hubspotFormTargetId = targetId || "conversion-touch-point-hubspot";
+
+    useEffect(() => {
+        // the Hubspot form requires jQuery object, we need to create a fake jQuery object to bypass this.
+        window.jQuery =
+            window.jQuery ||
+            function (nodeOrSelector: string | object) {
+                if (typeof nodeOrSelector == "string") {
+                    return document.querySelector(nodeOrSelector);
+                }
+                return nodeOrSelector;
+            };
+        if (window.hbspt?.isSuccessMessageShow) {
+            window.hbspt.isSuccessMessageShow = false;
+        }
+    }, []);
+
+    const onHubspotFormSubmitted = () => {
+        if (!window.hbspt.isSuccessMessageShow) {
+            onFormSubmitted?.();
+            onClose();
+            window.hbspt.isSuccessMessageShow = true;
+        }
+    };
+
+    const { formCreated } = useHubspotForm({
+        portalId: hubspotPortalId,
+        formId: hubspotFormId,
+        target: `#${hubspotFormTargetId}`,
+        locale: intl.locale.split("-").shift() as HubSpotFormLocale,
+        onFormSubmitted: onHubspotFormSubmitted,
+        onFormReady: ($form: IHubspotFormField[]) => {
+            setIsFormReady(true);
+            // populating the values for hidden fields
+            for (let i = 0; i < $form.length; i += 1) {
+                const inputField = $form[i];
+                if (!inputField) {
+                    continue;
+                }
+
+                const populatedValue = values[inputField.name];
+
+                if (populatedValue) {
+                    inputField.value = populatedValue;
+                }
+
+                if (selectedValue && inputField.value.indexOf(selectedValue) !== -1) {
+                    // Call click event to check the radio button on Hubspot form
+                    inputField.click();
+                }
+            }
+        },
+    });
+
+    return (
+        <>
+            {!formCreated && <LoadingComponent />}
+            {formCreated && (
+                <div className="conversion-touch-point-dialog-wrapper">
+                    {isFormReady && dialogTitle && <h3>{dialogTitle}</h3>}
+                    {/* After formCreated, the hubspot form will loaded in #get-in-touch-hubspot div. So it should be there.*/}
+                    <div id={hubspotFormTargetId} className="hubspot-form"></div>
+                    {/* The hubspot form doesn't include Cancel button. Adding the Cancel button to make Dialog consistent with other dialogs */}
+                    {isFormReady && showCancelButton && (
```

Comment:
I have one concern: if `showCancelButton` is set to false, it seems that there is no way for us to close the dialog, isn't it?
→ Is there any thing like a `x` icon to close the dialog?

Reply:
Yes, we have a close icon at the top right of the dialog.

Reply:
Then, should we call `onClose` as well when users click on the icon?

Reply:
yes, right.

Reply:
ok then. I've overlooked the trigger of `onClose` when users click on the icon.